### ISSUE TITLE
How does the "weight column"(weights for each row) passed on lgb.Dataset affects the "objective function"(log loss) in training?

### ISSUE THREAD

**Issue by NahuelGrasso** [microsoft/LightGBM#2549]
## Summary

How does the **weight column**(weights for each row) passed on lgb.Dataset affects the **objective function**(log loss) in training?
Does the objective function take into account the weights of each row or should we change the **objective function** to consider the **weights**?

## Motivation

I am trying to do a binary classification where I know that certain rows must be predicted right (either true or false) and thats the reason I want to set some higher weights to these observations. What I mean to do here is to give more importance in training to those rows I know more certainty they have to be right.
The problem I am facing is that I don't know if the objective function (I am using log loss) considers the weights in training or not. Or if I should customize the function and how.

## Description
I just took this example from the documentation (and I changed it a little bit):

### Create dataset for lightgbm
```
lgb_train = lgb.Dataset(X_train, y_train,
                        weight=W_train)
lgb_eval = lgb.Dataset(X_test, y_test, reference=lgb_train,
                       weight=W_test)
```

### Specify your configurations as a dict
```
params = {
    'boosting_type': 'gbdt',
    'objective': 'binary',
    'metric': 'binary_logloss',
    'num_leaves': 200,
    'learning_rate': 0.05,
    'feature_fraction': 0.8,
    'bagging_fraction': 0.8,
    'bagging_freq': 5,
    'verbose': 0
}
```
### Train
```
gbm = lgb.train(params,
                lgb_train,
                num_boost_round=2000,
                valid_sets=[lgb_train,lgb_eval],
                early_stopping_rounds=5, 
                verbose_eval=10)
```

**Comment by StrikerRUS** [comment]
@NahuelGrasso 
> The problem I am facing is that I don't know if the objective function (I am using log loss) considers the weights in training or not. 

It does.

https://github.com/microsoft/LightGBM/blob/7cfb492e7733f73526fbf8958d850bbb94af83ed/src/objective/binary_objective.hpp#L124-L136

Also, refer to https://github.com/microsoft/LightGBM/issues/2023#issuecomment-466633536 and #905.

**Comment by NahuelGrasso** [comment]
Thanks! :)
